src/main/java/org/mskcc/limsrest/service/PromoteBanked.java

```diff
@@ -336,6 +336,7 @@ public void createRecords(DataRecord bankedSampleRecord, DataRecord req, String
         String newIgoId = requestId + "_" + (maxExistentId + offset);
         try {
             DataRecord promotedSampleRecord = req.addChild("Sample", user);
+            List<DataRecord> missionBioProtocol1 = dataRecordManager.queryDataRecords("MissionBio Tapestri Library Prep Protocol 1s", "SampleId ='" + newIgoId + "'", user);
```

Comment:
Does `.queryDataRecords' create a new record if one doesn't already exist? If not, this won't return anything. We need to create a new `MissionBio Tapestri Library Prep Protocol 1s` record and add it as a child to the sample

Reply:
Correct. Addressed.